numpydoc/numpydoc.py

```diff
@@ -177,7 +178,7 @@ def mangle_signature(app, what, name, obj, options, sig, retann):
 
     if not hasattr(obj, '__doc__'):
         return
-    doc = get_doc_object(obj)
+    doc = get_doc_object(obj, config={'show_class_members': False})
```

Comment:
We just want the signature at this point, not all the side effects of parsing all the members of `obj`